### PR TITLE
Story3 detail page

### PR DESCRIPTION
**Pull request by maxwell-01**

### CODE REVIEW

File: planet.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+require 'vendor/autoload.php';
+
+include 'src/Planet.php';
+use PlanetApp\PlanetHydrator;
+
+$dsn = 'mysql:host=db; dbname=planet_collection';
+$username = 'root';
+$password = 'password';
+
+if(!isset($_GET['planetId'])){header('Location: ./index.php');}
```
Comment (norayna) [inline planet.php]: put the header on a new line

File: planet.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<?php
+require 'vendor/autoload.php';
+
+include 'src/Planet.php';
```
Comment (norayna) [inline planet.php]: why include instead of use?

Comment (norayna) [reply planet.php]: get rid

File: planet.php (hunk 1 above)
Comment (maxwell-01) [reply planet.php]: done

File: planet.php (hunk 2 above)
Comment (maxwell-01) [reply planet.php]: not needed at all, removed

File: planet.php (hunk 1 above)
Comment (norayna) [reply planet.php]: put this on line 6

Comment (maxwell-01) [reply planet.php]: great suggestion - will improve redirect speed

File: planet.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+<body>
+<main>
+    <a class="back-button" href="./index.php#planet-<?=$planetId?>">&#8637 <object class="desktop-button-text"> Planets</object></a>
+    <h1><?=$planet->getName()?></h1>
+    <div class="planet-image-section">
+        <img src="./images/<?=$planet->getImageSrc()?>" alt="<?=$planet->getName()?>">
+        <p class="image-source-text">Source: <?=$planet->getImageDescription()?></p>
+    </div>
+    <div class="planet-details-section">
+        <?php if($planet->getFullName() != null) {echo '<h2>Full Name: '.$planet->getFullName().'</h2>';}?>
```
Comment (norayna) [inline planet.php]: null coalesce instead? `$planet->getFullName() ?? ....`

File: planet.scss [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  font-size: 1.6rem;
+  text-align: center;
+}
+
+main {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  font-size: 1.5rem;
+  margin: auto;
```
Comment (norayna) [inline planet.scss:25]: what does this do?

File: planet.scss [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    margin: 5px;
+    transition: 1s;
+    text-decoration: none;
+    &:hover {
+      border-color: #ffffff;
+      color: #ffffff;
+    }
+  }
+
+  .desktop-button-text {
```
Comment (norayna) [inline planet.scss]: desktop-back-button-text

File: planet.scss [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    position: absolute;
+    top: 20px;
+    left: 20px;
+    background-color: #000000;
+    color: #03FC13;
+    font-weight: 600;
+    font-size: 2rem;
+    border: solid 1px #03FC13;
+    padding: 20px 23px;
+    margin: 5px;
```
Comment (norayna) [inline planet.scss]: instead increase top and left properties and have no margin

File: planet.php (hunk 3 above)
Comment (maxwell-01) [reply planet.php]: replaced with ternary

File: planet.scss (hunk 5 above)
Comment (maxwell-01) [reply planet.scss]: done

File: planet.scss (hunk 6 above)
Comment (maxwell-01) [reply planet.scss]: done

File: planet.scss [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+  .detail-table-heading {
+    font-weight: 600;
+    color: #E875AB;
+  }
+}
+
+@media screen and (min-width: 800px) {
+
+  main {
```
Comment (norayna) [inline planet.scss]: try to center the main vertically

Comment (maxwell-01) [reply planet.scss]: done

File: planet.php [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        <div class="planet-details-section">
+            <?=($planet->getFullName() != null) ? '<h2>Full Name: '.$planet->getFullName().'</h2>':'';?>
+            <p class="planet-description-text">
+                <?=$planet->getDescription()?>
+            </p>
+            <div class="details-table">
+                <p class="detail-table-heading">Planet type</p>
+                <p><?=$planet->getType()?></p>
+                <p class="detail-table-heading">Length of day</p>
+                <p><?=$planet->getLengthOfDay()==='unknown'?'unknown':$planet->getLengthOfDay().' Earth days'?></p>
```
Comment (norayna) [inline planet.php]: getter should return null rather than 'unknown'

File: planet.php [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+                <p class="detail-table-heading">Length of year</p>
+                <p><?=$planet->getLengthOfYear()?> Earth years</p>
+                <p class="detail-table-heading">Mean radius</p>
+                <p><?=$planet->getMeanRadiusKm()?>km</p>
+                <p class="detail-table-heading">Mean distance from Sun</p>
+                <p><?=$planet->getMeanDistanceFromSunAu()?>au</p>
+                <p class="detail-table-heading">Star</p>
+                <p><?=$planet->getSun()?></p>
+                <p class="detail-table-heading">Mass</p>
+                <p><?=$planet->getMassEarths() === 'unknown'?'unknown':$planet->getMassEarths().' Earths'?></p>
```
Comment (norayna) [inline planet.php]: getter should return null rather than 'unknown'

**Review by norayna — APPROVED** [review]
APPROVED!

**Pull request merged**